### ISSUE TITLE
use more stable expit

### ISSUE THREAD

**Issue by jasmainak** [pavanramkumar/pyglmnet#102]
`scikit-learn` computes `expit` in a slightly different way. I'm wondering how much difference this creates: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/fixes.py#L44-64. We should explore this as `expit` is used in many places. Also, if we compute it this way, we don't even need `scipy` as dependency !

**Comment by pavanramkumar** [comment]
interesting! 

i remember learning about how fast `expit` is in relation to `1/(1+np.exp(-x))` when i started implementing it and i'm surprised that sklearn doesn't use it.

So I tried the following:

```python
import numpy as np
x = np.random.randn(1000)
```

```python
from scikit.sparse import expit
%timeit y = expit(x)
```
The slowest run took 16.24 times longer than the fastest. This could mean that an intermediate result is being cached. 
100000 loops, best of 3: 18.8 µs per loop

```python
%timeit y = (1+np.tanh(x/2))/2
```

The slowest run took 13.75 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 28.2 µs per loop

**Comment by pavanramkumar** [comment]
seems like `scipy` 1 `sklearn` 0!

**Comment by jasmainak** [comment]
yeah, I explored this issue a bit more and it appears that indeed the `scikit-learn` version is slower. The only caveat is that you need `scipy > 0.14` because in the older version, it was returning `nan` due to the fact that `np.exp(x)` was `inf` when there was an overflow. It seems to have been fixed now, so if we are using `scipy > 0.14` we should be fine.

ps: It seems the `sklearn` version performs comparably well if you do `x = 0.1*np.abs(np.random.randn(1000))`. Anyway, I was worried more about stability than speed and I think that's fixed now ...
